Source/Core/Core/PowerPC/JitCommon/JitCache.cpp

```diff
@@ -347,12 +330,19 @@ void JitBaseBlockCache::MoveBlockIntoFastCache(u32 addr, u32 msr)
   }
   else
   {
-    FastLookupEntryForAddress(addr) = block;
+    // Drop old icache entry
+    if (iCache[block->in_icache] == block)
+      iCache[block->in_icache] = nullptr;
+
+    // And create a new one
+    size_t icache = FastLookupEntryForAddress(addr);
+    iCache[icache] = block;
+    block->in_icache = icache;
     LinkBlock(*block);
   }
 }
 
-JitBlock*& JitBaseBlockCache::FastLookupEntryForAddress(u32 address)
+size_t JitBaseBlockCache::FastLookupEntryForAddress(u32 address)
```

Comment:
The name of this method sounds a little odd. The old one actually did a lookup and returned an item, but this one returns...an index, key, offset, something like that?